01-syntax/03-data-structures/deep-comparation/deepEqual.js

```diff
@@ -4,9 +4,39 @@
  */
 /* global deepEqual */
 
-// → Your code here.
+var obj;
 
-var obj = {
+function deepEqual(item1, item2) {
+  var key1;
+  var key2;
+  var match;
+  if (typeof item1 === 'object' && typeof item2 === 'object' && item1 && item2) {
```

Comment:
What about providing helper function `isUnderfined` ? it may help you to have  more readable code